crates/solver/src/settlement_ranker.rs

```diff
@@ -151,3 +152,67 @@ impl SettlementRanker {
         Ok((rated_settlements, errors))
     }
 }
+
+fn compare_solutions(lhs: &RatedSettlement, rhs: &RatedSettlement, decimals: i32) -> Ordering {
+    let precision = BigRational::from_i8(10).unwrap().pow(decimals);
+    let rounded_lhs = (lhs
+        .objective_value()
+        .checked_div(&precision)
+        .expect("precision cannot be 0"))
+    .floor();
```

Comment:
nit: the extra parenthesis are weird and I think unnecessary:
```suggestion
    let rounded_lhs = lhs
        .objective_value()
        .checked_div(&precision)
        .expect("precision cannot be 0")
        .floor();
```